molecule/provisioner/ansible/playbooks/docker/create.yml

```diff
@@ -27,7 +27,7 @@
       register: platforms
 
     - name: Discover local Docker images
-      docker_image_facts:
+      docker_image_info:
```

Comment:
How do we maintain compat with previous versions of Ansible?

Please see failures on the CI: https://travis-ci.com/ansible/molecule/jobs/200941048#L603.

Do we need some conditional logic here?

Reply:
I'll investigate and fix.

Reply:
So I misread the Ansible docs about when `docker_image_info` was introduced (it's 2.8 not 2.1). I'll probably try a conditional, since I don't believe we can set and use a module reference within a variable.

Reply:
> So I misread the Ansible docs about when `docker_image_info` was introduced (it's 2.8 not 2.1). I'll probably try a conditional, since I don't believe we can set and use a module reference within a variable.

I think you can try this condition 
```yaml
- name: Discover local Docker images
  docker_image_facts:
    name: "molecule_local/{{ item.item.name }}"
    docker_host: "{{ item.item.docker_host | default(lookup('env', 'DOCKER_HOST') or 'unix://var/run/docker.sock') }}"
  with_items: "{{ platforms.results }}"
  when: 
    - not item.pre_build_image | default(false)
    - ansible_version.full is version('2.8.0', '<')
  register: docker_images

- name: Discover local Docker images
  docker_image_info:
    name: "molecule_local/{{ item.item.name }}"
    docker_host: "{{ item.item.docker_host | default(lookup('env', 'DOCKER_HOST') or 'unix://var/run/docker.sock') }}"
  with_items: "{{ platforms.results }}"
  when: 
    - not item.pre_build_image | default(false)
    - ansible_version.full is version('2.8.0', '>=')
  register: docker_images
`

Reply:
> ...I don't believe we can set and use a module reference within a variable.

If I'm understanding this correctly, then I think you can, using a ternery and `action`/`args` in the task body, but it's not the prettiest-looking ansible, and requires both modules to take the same inputs and produce the same outputs. It's also not the prettiest ansible:

```yaml
# Warning! While this is based on working code, I haven't tested this particular snippet.
- name: Determine which docker image info module to use
  set_fact:
    _docker_image_info_module: >-
      {{ ansible_version.full is version_compare('2.8', '>=') |
         ternary('docker_image_info', 'docker_image_facts') }}

- name: Discover local Docker images
  action: "{{ _docker_image_info_module }}"
  args:
    name: "molecule_local/{{ item.item.name }}"
    docker_host: "{{ item.item.docker_host | default(lookup('env', 'DOCKER_HOST') or 'unix://var/run/docker.sock') }}"
  with_items: "{{ platforms.results }}"
  when: 
    - not item.pre_build_image | default(false)
  register: docker_images
```

Reply:
Managed to find a fix for this? I hope we are not going towards branches for each ansible version, it would be a maintenance hell.